### DISCUSSION TITLE
How does thruster file Direct work.

### DISCUSSION THREAD

**Discussion by prashant19012002** [JSBSim-Team/jsbsim#972] [Q&A]
For example wkt if Ct and Cp of a propeller is mentioned wrt to advance ratio then JSBSim calculates the thrust for any conditions based on this values. But how does the file direct work .
![image](https://github.com/JSBSim-Team/jsbsim/assets/145685228/e4f724f8-f030-4798-994d-2cf4d56f3f3b)
![image](https://github.com/JSBSim-Team/jsbsim/assets/145685228/8d02c21a-2502-4dc6-afcb-47eb0471991f)

**Comment by seanmcleod** [comment]
The `Direct` option is used as the comment mentions for engine types that compute the thrust, e.g. a gas turbine engine like this one in the 737 model, i.e. there is no propeller that takes the power and torque from the engine to then produce the thrust.

https://github.com/JSBSim-Team/jsbsim/blob/ae4652a268d0475dbf93d47a32ab7ddf4120aaef/aircraft/737/737.xml#L124-L140

P.S
Rather than posting screenshots for your examples above, rather paste them inside a triple back tick set marked as xml. That way the text is searchable in future etc.

**Reply by prashant19012002** [reply]
I saw the engine file CFM56 since its a turbine engine the trust values are mentioned based on density altitude and mach no. .
But can i know how do we modify it for electric UAV as in this example of [x8](https://github.com/AOS55/Fixedwing-Airsim/blob/master/x8/electric800W.xml) UAV . 

`<?xml version="1.0"?>

<electric_engine name="electric">
 
 <power unit="WATTS">  8.0 </power>
</electric_engine>`




I tried to run this x8 plane with some cruise scripts and i could do this and achieve trim based on your discussions in thread #853 . Now i wanted to replicate the same thing in for my electric motor propeller combo (but instead of a separate propeller i wanted to mention thrust directly). 
The main reason for doing this is for some reason i am not able to start my engine in normal case where i mention propeller  and engine file.
